src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -76,65 +79,23 @@ public PrometheusMeteringController(ApplicationClock clock, PrometheusMetricsPro
-                    throw new MeteringException(
-                        String.format("Unable to fetch OpenShift metrics: %s", metricData.getError())
-                    );
-                }
+                // Reset the start/end dates to ensure they span a complete hour.
+                // NOTE: If the prometheus query step changes, we will need to adjust this.
+                OffsetDateTime queryStart = clock.startOfHour(start);
+                OffsetDateTime queryEnd = clock.endOfHour(end);
 
-                // Given the possibility of a very large number of events, we will batch persist them
-                // to provide the ability to tweak them.
-                List<Event> events = new LinkedList<>();
-                int eventCount = 0;
-                for (QueryResultDataResult r : metricData.getData().getResult()) {
-                    Map<String, String> labels = r.getMetric();
-                    String clusterId = labels.get("_id");
-                    String sla = labels.get("support");
-                    String usage = labels.get("usage");
-
-                    // For the openshift metrics, we expect our results to be a 'matrix'
-                    // vector [(instant_time,value), ...] so we only look at the result's getValues()
-                    // data.
-                    for (List<BigDecimal> measurement : r.getValues()) {
-                        BigDecimal time = measurement.get(0);
-                        BigDecimal value = measurement.get(1);
-
-                        Event event = MeteringEventFactory.openShiftClusterCores(account, clusterId,
-                            sla, usage, clock.dateFromUnix(time), value.doubleValue());
-                        events.add(event);
-                        eventCount++;
-
-                        if (events.size() >= openshiftProperties.getEventBatchSize()) {
-                            log.info("Saving {} events", events.size());
-                            eventController.saveAll(events);
-                            events.clear();
-                        }
-                    }
+                log.info("Collecting labels for OpenShift metrics...");
+                Map<LabelSetKey, Map<String, String>> labels = getSubscriptionLabels(account, queryStart,
+                    queryEnd);
+                if (labels.isEmpty()) {
+                    log.info("No subscription labels found. No events will be generated.");
                 }
 
-                // Flush the remainder
-                if (!events.isEmpty()) {
-                    log.debug("Saving remaining events: {}", events.size());
-                    eventController.saveAll(events);
-                }
-                log.info("Created {} Events for OpenShift metrics.", eventCount);
+                log.info("Collecting OpenShift metrics and generating events");
+                int eventCount = generateEventsFromMetrics(labels, account, queryStart, queryEnd);
```

Comment:
Wondering if the log statement above this should be the first line of the `generateEventsFromMetrics` method to ensure that the message always goes with its logic